e2e_playwright/st_experimental_fragment_basics_test.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import wait_for_app_run
+
+
+def get_uuids(app: Page):
+    text_in_fragment = app.get_by_test_id("stMarkdown").first.text_content()
+    text_outside_fragment = app.get_by_test_id("stMarkdown").last.text_content()
```

Comment:
I think this will probably work fine. But there is also a little bit of danger of flakiness since it might not be 100% guaranteed that the text is already in the DOM. But since we are always waiting for the app run to finish, I would assume this works fine. 

And alternative might be to just use a predicated counter (counting the number of reruns) instead of a uuid. And then we can actually wait (via `expect().to_have_text()`) for the expected count.

Reply:
I think this is likely fine, but to be safe I added an `expect` call that waits for two markdown elements to appear before grabbing their text